modules/codegen/test/src/smithy4s/codegen/internals/ScaladocSpec.scala

```diff
@@ -0,0 +1,109 @@
+package smithy4s.codegen.internals
+
+final class ScaladocSpec extends munit.FunSuite {
+  import TestUtils._
+
+  test("Generate Scaladoc - include struct field docs") {
+    val smithy =
+      """
+        |$version: "2"
+        |
+        |namespace smithy4s
+        |
+        |/// Struct docs
+        |structure DocumentedStruct {
+        |  /// int docs
+        |  @required
+        |  int: Integer
+        |  /// struct docs
+        |  struct: UndocumentedStruct
+        |  
+        |  @required
```

Comment:
Added this to highlight this problem (maybe it's just smithy synax rules in play?).
If the ordering of annotation and comment is like here (first annotation and then docs), then in AWS model, the documentation trait is missing, only required trait is there. I think the same applies for other annotations, like `@packedInputs` before the comment on an operation/service etc.